frontend/ui/widget/htmlboxwidget.lua

```diff
@@ -196,4 +214,36 @@ function HtmlBoxWidget:onHoldReleaseText(callback, ges)
     return true
 end
 
+function HtmlBoxWidget:getLinkByPosition(pos)
+    local page = self.document:openPage(self.page_number)
+    local links = page:getPageLinks()
+    page:close()
+
+    for _, link in pairs(links) do
+        if pos.x >= link.x0 and pos.x < link.x1 and pos.y >= link.y0 and pos.y < link.y1 then
+            return link
```

Comment:
Never really had PDF files with links, so just curious: how does MuPDF do when a link spans multiple words (may not happen with dictionaries definitions if they are a single word), and the line is wrapped in the middle of these multiple words: does it return 2 links (one for the first word at the far right of 1st line, one for the second word at the far lest of 2nd line)? (If not, the above may not work in such cases)

Reply:
Most links I come across in regular use are footnotes (i.e., just a few digits) but should be simple to check on the vast majority of PDFs. TOCs are common.

Reply:
This PDF has a multiline link at the bottom of p. 471 http://www.oapen.org/search?identifier=641735 (*Sémantique formelle: Volume 1 : Introduction à la grammaire de Montague*)

Reply:
MuPDF returns one link per line if the link spans multiple lines.